server/src/archiver/disk.rs

```diff
@@ -44,20 +47,28 @@ impl Archiver for DiskArchiver {
         &self,
         files: &[&str],
         base_directory: Option<String>,
-    ) -> Result<(), ServerError> {
+    ) -> Result<(), ServerArchiverError> {
         debug!("Archiving files on disk: {:?}", files);
         for file in files {
             debug!("Archiving file: {file}");
             let source = Path::new(file);
             if !source.exists() {
-                return Err(ServerError::FileToArchiveNotFound(file.to_string()));
+                return Err(ServerArchiverError::FileToArchiveNotFound {
+                    file_path: file.to_string(),
+                });
             }
 
             let base_directory = base_directory.as_deref().unwrap_or_default();
             let destination = Path::new(&self.config.path).join(base_directory).join(file);
             let destination_path = destination.to_str().unwrap_or_default().to_owned();
-            fs::create_dir_all(destination.parent().unwrap()).await?;
-            fs::copy(source, destination).await?;
+            fs::create_dir_all(destination.parent().unwrap())
```

Comment:
Hmm Im not sure if I like what does trace register. I would like those traces to include abstraction layer where we are in the system at the time of the error propagation/catch. So in this case, it would be something akin to this `ARCHIVER - failed to create file: {file} at path: {path} with error: {err}`. In the higher levels of the system, when propagating the error further I would attach more transient information, such as identifiers etc.. and not include the error at hand itself in the trace.